lib/alchemy/resources_helper.rb

```diff
@@ -158,5 +158,15 @@ def render_resources
     rescue ActionView::MissingTemplate
       render :partial => 'resource', :collection => resources_instance_variable
     end
+
+    # Returns all the params necessary to get you back from where you where
+    # before: the Ransack query and the current page.
+    #
+    def current_location_params
+      {
+        q: params[:q],
+        page: params[:page]
```

Comment:
I think that's even a good thing: If they delete a record, they'll be able to confirm the record's gone. So the only thing where they'd be a bit stumped is if they happen to edit the field that's being sorted on, and all of the sudden the record they edited is on page 1 instead of six - but that's an edge case they'll have to live with :imp: